airflow/jobs/base_job.py

```diff
@@ -93,7 +93,7 @@ def __init__(
 
     @classmethod
     @provide_session
-    def most_recent_job(cls, session: Session) -> Optional['BaseJob']:
+    def most_recent_job(cls, session=None) -> Optional['BaseJob']:
```

Comment:
Did we need to change this to `None`? This is  quirk between type checkers and docs here. :/

Reply:
Unfortunately yes, the below line calls it without a session.

https://github.com/apache/airflow/blob/aa8a30099884efbf6fd5fb276f3dc6a310866aba/airflow/www/views.py#L184

Reply:
I mean it looks like it does, but the decorator takes care of that.

So interaction between our provide_session and typecheckers then.